### PR TITLE
[release/7.0-rc1] net7.0 + net6.0 workload multitargeting changes

### PR DESCRIPTION
**Pull request by lewing**

### CODE REVIEW

**Comment by radical** [comment]
The wasm workload tests are passing now.

**Comment by lewing** [comment]
@dsplaisted  @marcpopMSFT 

```
      [] /datadisks/disk1/work/93E60831/w/A8A8094E/e/dotnet-net7/sdk/7.0.100-rc.2.22419.24/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/datadisks/disk1/work/93E60831/w/A8A8094E/e/blz_aot_prj_file_Release/blz_aot_prj_file_Release.csproj]
      [] /datadisks/disk1/work/93E60831/w/A8A8094E/e/dotnet-net7/sdk/7.0.100-rc.2.22419.24/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(448,5): error NETSDK1109: Runtime list file '/datadisks/disk1/work/93E60831/w/A8A8094E/e/dotnet-net7/packs/Microsoft.NETCore.App.Runtime.Mono.browser-wasm/data/RuntimeList.xml' was not found. Report this error to the .NET team here: https://aka.ms/dotnet-sdk-issue. [/datadisks/disk1/work/93E60831/w/A8A8094E/e/blz_aot_prj_file_Release/blz_aot_prj_file_Release.csproj]
```

**Comment by lewing** [comment]
@marcpopMSFT could it be https://github.com/dotnet/sdk/commit/0bc1ff8b3521150ce9f115d21c6dfd2a443c1043

**Comment by lewing** [comment]
Found the issue on our side, the tests were including a bunch of outdated logic that was breaking things

**Comment by radical** [comment]
Approved, assuming the CI is green.

**Review by radical — APPROVED** [review]
Assuming the ci is green

File: src/tests/BuildWasmApps/Wasm.Build.Tests/BuildEnvironment.cs [hunk 1]
```diff
@@ -112,6 +112,12 @@ public BuildEnvironment()
             // helps with debugging
             EnvVars["WasmNativeStrip"] = "false";
 
+            // Works around an issue in msbuild due to which
+            // second, and subsequent builds fail without any details
+            // in the logs
+            EnvVars["DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER"] = "1";
+            DefaultBuildArgs += " /nr:false";
```
Comment (akoeplinger) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/BuildEnvironment.cs:119]: do we have an issue filed for this?

**Pull request merged**